evalml/tests/component_tests/test_oversamplers.py

```diff
@@ -0,0 +1,241 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from evalml.pipelines.components import (
+    SMOTENCSampler,
+    SMOTENSampler,
+    SMOTESampler
+)
+from evalml.pipelines.components.utils import make_balancing_dictionary
+
+im = pytest.importorskip('imblearn.over_sampling', reason='Skipping test because imbalanced-learn not installed')
+
+
+binary = pd.Series([0] * 800 + [1] * 200)
+multiclass = pd.Series([0] * 800 + [1] * 150 + [2] * 50)
+
+
+@pytest.mark.parametrize("y,sampling_ratio,result",
+                         [(binary, 1, {0: 800, 1: 800}),
+                          (binary, 0.5, {0: 800, 1: 400}),
+                          (binary, 0.25, {0: 800, 1: 200}),
+                          (binary, 0.1, {0: 800, 1: 200}),
+                          (multiclass, 1, {0: 800, 1: 800, 2: 800}),
+                          (multiclass, 0.5, {0: 800, 1: 400, 2: 400}),
+                          (multiclass, 0.25, {0: 800, 1: 200, 2: 200}),
+                          (multiclass, 0.1, {0: 800, 1: 150, 2: 80})])
+def test_make_balancing_dictionary(y, sampling_ratio, result):
+    dic = make_balancing_dictionary(y, sampling_ratio)
+    assert dic == result
+
+
+@pytest.mark.parametrize("sampler", [SMOTESampler, SMOTENCSampler, SMOTENSampler])
+def test_init(sampler):
+    parameters = {
+        "sampling_ratio": 0.5,
+        "sampling_ratio_dict": {},
+        "k_neighbors": 2,
+        "n_jobs": -1
+    }
+    if 'SMOTENC' in sampler.name:
+        parameters['categorical_features'] = [0]
+    oversampler = sampler(**parameters)
+    assert oversampler.parameters == parameters
+
+
+@pytest.mark.parametrize("sampler", [SMOTESampler(sampling_ratio=1),
+                                     SMOTENCSampler(sampling_ratio=1, categorical_features=[0]),
+                                     SMOTENSampler(sampling_ratio=1)])
+def test_none_y(sampler):
+    X = pd.DataFrame({"a": [i for i in range(5)],
+                      "b": [1 for i in range(5)]})
+    oversampler = sampler
+    with pytest.raises(ValueError, match="y cannot be none"):
+        oversampler.fit(X, None)
+    with pytest.raises(ValueError, match="y cannot be none"):
+        oversampler.fit_transform(X, None)
+    oversampler.fit(X, pd.Series([0] * 4 + [1]))
+    oversampler.transform(X, None)
+
+
+@pytest.mark.parametrize("sampler", [SMOTESampler(sampling_ratio=1),
+                                     SMOTENCSampler(sampling_ratio=1, categorical_features=[0]),
+                                     SMOTENSampler(sampling_ratio=1)])
+@pytest.mark.parametrize("data_type", ["np", "pd", "ww"])
+def test_no_oversample(data_type, sampler, make_data_type, X_y_binary):
```

Comment:
What is this one testing?

Reply:
This is just making sure we don't oversample when we don't need to